--- a/src/test/java/io/papermc/paper/permissions/MinecraftCommandPermissionsTest.java
+++ b/src/test/java/io/papermc/paper/permissions/MinecraftCommandPermissionsTest.java
@@ -79,6 +79,10 @@
         "minecraft.command.gamemode.survival.other",
         // Purpur end - Skip junit tests for purpur commands
 
+        // Zero start - Skip junit tests for zero commands
+        "minecraft.command.settps",
+        "minecraft.command.zero",
+        // Zero end - Skip junit tests for zero commands
+        
         "minecraft.command.selector"
     );